libindy/src/commands/non_secrets.rs

```diff
@@ -303,6 +366,256 @@ impl NonSecretsCommandExecutor {
         Ok(res)
     }
 
+    fn get_schema(&self,
+                  pool_handle: PoolHandle,
+                  wallet_handle: WalletHandle,
+                  submitter_did: &str,
+                  id: &str,
+                  options_json: &str,
+                  cb: Box<Fn(IndyResult<String>) + Send>) {
+        trace!("get_schema >>> pool_handle: {:?}, wallet_handle: {:?}, submitter_did: {:?}, id: {:?}, options_json: {:?}",
+               pool_handle, wallet_handle, submitter_did, id, options_json);
+
+        let options = try_cb!(serde_json::from_str::<GetCacheOptions>(options_json).to_indy(IndyErrorKind::InvalidStructure, "Cannot deserialize options"), cb);
+
+        let cache = if !options.no_cache.unwrap_or(false) {
+            let options_json = json!({
+                "retrieveType": false,
+                "retrieveValue": true,
+                "retrieveTags": true,
+            }).to_string();
+            match self.wallet_service.get_record(wallet_handle, SCHEMA_CACHE, id, &options_json) {
+                Ok(record) => Ok(Some(record)),
+                Err(err) => if err.kind() == IndyErrorKind::WalletItemNotFound {Ok(None)} else {Err(err)}
+            }
+        } else {Ok(None)};
+        let cache = try_cb!(cache, cb);
+
+        if cache.is_some() {
+            let cache = cache.unwrap();
+            let min_fresh = options.min_fresh.unwrap_or(-1);
+            if min_fresh >= 0 {
+                let ts = SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_secs() as i32;
```

Comment:
unwrap here may fail in theory on windows, isn't it? In this case I suggest to handle the error (may be as invalid state or even ignore somehow, not sure)